src/index.js

```diff
@@ -26,10 +29,21 @@ module.exports = postcss.plugin('postcss-modules', (opts = {}) => {
   }
 
   return (css, result) => {
+    const inputFile = css.source.input.file;
+
     const resultPlugins = result.processor.plugins
       .filter(plugin => plugin.postcssPlugin !== 'postcss-modules');
 
-    const plugins = [...defaultPlugins[scopeBehaviour], ...resultPlugins];
+    let pluginList = defaultPlugins[defaultScopeBehaviour];
+
+    if (globalModulePathWhitelist) {
+      const isGlobalModule = globalModulePathWhitelist
+        .filter(regex => inputFile.match(regex))
+        .length !== 0;
+      pluginList = defaultPlugins[isGlobalModule ? 'global' : 'local'];
```

Comment:
Not quite sure I understand what you're saying.. since I reference that in `defaultScopeBehaviour` when the plugin loads it should not be modifiable. Or are there situations where plugins need to be able to mutate `opts` on the fly via some reference they get a handle on?